packages/contracts/contracts/TransactionManager.sol

```diff
@@ -503,11 +499,22 @@ contract TransactionManager is ReentrancyGuard, ProposedOwnable, ITransactionMan
       // on the sending chain
       require(msg.sender == txData.router, "#F:016");
 
+      // Validate the user has signed
+      require(recoverFulfillSignature(txData.transactionId, relayerFee, txData.receivingChainId, txData.receivingChainTxManagerAddress, signature) == txData.user, "#F:022");
```

Comment:
I'm confused why this has been moved out from under the 'validation scope' above - although I'm not familiar enough with code conventions around sol to know why. do you mind explaining? what is that 'validation scope' exactly? if this is a validation step, shouldn't it also be under that scope?

Reply:
It's moved out of the combined validation scope to strengthen the domain separation enforcement. You can see now we do the same recovery on both chains, but on the sending chain we use what's in the transaction data, whereas on the receiving chain we use what we can pull from the chain itself. the 'validation scope' is actually just a trick around the stack too deep issues -- by placing things in brackets you create a new scope that doesn't impact the local stack. i grouped all validation there where it should be performed the exact same across both chains

Reply:
Ah understood, thanks for explaining!